x/stakeibc/keeper/update_redemption_rates_test.go

```diff
@@ -3,125 +3,111 @@ package keeper_test
 	_ "github.com/stretchr/testify/suite"
 
+	minttypes "github.com/Stride-Labs/stride/v8/x/mint/types"
+
 	recordtypes "github.com/Stride-Labs/stride/v8/x/records/types"
 
 	stakeibctypes "github.com/Stride-Labs/stride/v8/x/stakeibc/types"
 )
 
-type UpdateRedemptionRatesTestCase struct {
-	hostZone   stakeibctypes.HostZone
-	allRecords []recordtypes.DepositRecord
+// TODO: Use array-of-test-cases setup for this redemption rate test
+// TODO [LSM]: Fix randomized test
+// TODO [LSM]: Add balanced/unbalanced delegation test cases
+// TODO [LSM]: Add test cases with multiple deposit records
+// TODO [LSM]: Add unit tests for sub-functions
+
+type UpdateRedemptionRateTestCase struct {
+	balancedDelegation    sdkmath.Int
+	unbalancedDelegation  sdkmath.Int
+	undelegatedBal        sdkmath.Int
+	justDepositedBal      sdkmath.Int
+	stSupply              sdkmath.Int
+	initialRedemptionRate sdk.Dec
+}
+
+// Helper function to look up the redemption rate and check it against expectations
+func (s *KeeperTestSuite) checkRedemptionRateAfterUpdate(expectedRedemptionRate sdk.Dec) {
+	hostZone, found := s.App.StakeibcKeeper.GetHostZone(s.Ctx, HostChainId)
+	s.Require().True(found, "host zone should have been found but was not")
+	s.Require().Equal(expectedRedemptionRate, hostZone.RedemptionRate, "redemption rate")
 }
 
-func (s *KeeperTestSuite) SetupUpdateRedemptionRates(
-	balancedDelegation sdkmath.Int,
-	undelegatedBal sdkmath.Int,
-	justDepositedBal sdkmath.Int,
-	stSupply sdkmath.Int,
-	initialRedemptionRate sdk.Dec,
-) UpdateRedemptionRatesTestCase {
+func (s *KeeperTestSuite) SetupUpdateRedemptionRates(tc UpdateRedemptionRateTestCase) []recordtypes.DepositRecord {
 	// add some deposit records with status STAKE
 	//    to comprise the undelegated delegation account balance i.e. "to be staked"
 	toBeStakedDepositRecord := recordtypes.DepositRecord{
-		HostZoneId: "GAIA",
-		Amount:     undelegatedBal,
+		HostZoneId: HostChainId,
+		Amount:     tc.undelegatedBal,
 		Status:     recordtypes.DepositRecord_DELEGATION_QUEUE,
 	}
 	s.App.RecordsKeeper.AppendDepositRecord(s.Ctx, toBeStakedDepositRecord)
 
 	// add a balance to the stakeibc module account (via records)
 	//    to comprise the stakeibc module account balance i.e. "to be transferred"
 	toBeTransferedDepositRecord := recordtypes.DepositRecord{
-		HostZoneId: "GAIA",
-		Amount:     justDepositedBal,
+		HostZoneId: HostChainId,
+		Amount:     tc.justDepositedBal,
 		Status:     recordtypes.DepositRecord_TRANSFER_QUEUE,
 	}
 	s.App.RecordsKeeper.AppendDepositRecord(s.Ctx, toBeTransferedDepositRecord)
 
-	// set the stSupply by minting to a random user account
-	user := Account{
-		acc:           s.TestAccs[0],
-		stAtomBalance: sdk.NewCoin(StAtom, stSupply),
-	}
-	s.FundAccount(user.acc, user.stAtomBalance)
+	// set the stSupply by minting
+	supply := sdk.NewCoins(sdk.NewCoin(StAtom, tc.stSupply))
+	s.App.BankKeeper.MintCoins(s.Ctx, minttypes.ModuleName, supply)
```

Comment:
Who are these minted to, just the module acct?

Reply:
yeah they don't need to be sent to a specific user, all that matter's is the supply increases